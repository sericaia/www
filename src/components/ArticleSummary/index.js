import React from 'react'
import PropTypes from 'prop-types'
import { Link } from 'gatsby'

import Icons from '../Icons'

const ArticleSummary = ({ link, title, date, icons, excerpt }) => (
  <div>
    <h3
      style={{
        marginBottom: '0.3rem',
      }}
    >
      <Link
        to={link}
        style={{
          textDecoration: 'none',
          color: '#333',
          fontWeight: 'bold',
          fontSize: '1.2rem',
        }}
      >
        <Icons icons={icons} />
        {title}
      </Link>
    </h3>
    <p
      style={{
        color: '#767676',
        marginBottom: '0.3rem',
        fontWeight: 400,
        lineHeight: '1.1',
      }}
    >
      {date}
    </p>
    <p
      style={{
        fontSize: '0.9rem',
        fontStyle: 'italic',
      }}
    >
      {excerpt}
    </p>
  </div>
)

ArticleSummary.propTypes = {
  link: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
  date: PropTypes.string.isRequired,
  icons: PropTypes.arrayOf(PropTypes.string),
  excerpt: PropTypes.string.isRequired,
}

export default ArticleSummary
