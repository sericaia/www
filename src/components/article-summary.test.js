import React from 'react'
import { render } from '@testing-library/react'
import ArticleSummary from './article-summary'

describe('ArticleSummary', () => {
  it('renders correctly', () => {
    const props = {
      link: '/summary',
      title: 'Fake article',
      date: 'January 21, 2020',
      excerpt: 'lorem ipsum',
    }
    const { container } = render(<ArticleSummary {...props} />)
    expect(container).toMatchSnapshot()
  })
})
