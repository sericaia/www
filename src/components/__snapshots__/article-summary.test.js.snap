// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleSummary renders correctly 1`] = `
<div>
  <div>
    <h3
      style="margin-bottom: 0.3rem;"
    >
      <a
        href="/summary"
        style="text-decoration: none; color: rgb(51, 51, 51); font-weight: bold; font-size: 1.2rem;"
      >
        Fake article
      </a>
    </h3>
    <h4
      style="color: grey; margin-bottom: 0.3rem; font-weight: 400;"
    >
      January 21, 2020
    </h4>
    <p
      style="font-size: 0.9rem; font-style: italic;"
    >
      lorem ipsum
    </p>
  </div>
</div>
`;
